builtin/logical/pki/crl_util.go

```diff
@@ -1266,12 +1261,15 @@ func buildCRL(sc *storageContext, crlInfo *crlConfig, forceNew bool, thisIssuerI
 		// forceNew option). In previous versions of Vault (1.10 series and
 		// earlier), we'd have queried the certs below, whereas we now have
 		// an assignment from a pre-queried list.
-		goto WRITE
+	} else {
+		revokedCerts = revoked
 	}
 
-	revokedCerts = revoked
+	crlLifetime, err := time.ParseDuration(crlInfo.Expiry)
```

Comment:
This wouldn't fix the issue, right? It happens in the same control flow as the earlier code change, just in a different order.

I believe the solution I was suggesting was to add it here, which you've done below:

https://github.com/hashicorp/vault/blob/0423ffbe241b19bbd5effd73ecc021a3fdf4c3e9/builtin/logical/pki/storage.go#L1152-L1178

In particular, the bug is that previous versions of Vault allowed empty strings (if the CRL config was never written), but we now strictly enforce not having that.

So we should replace those empty strings with the default values.

In particular the test you point out here:

https://github.com/hashicorp/vault/blob/0423ffbe241b19bbd5effd73ecc021a3fdf4c3e9/builtin/logical/pki/crl_test.go#L948-L960

works, as it should, but only when a Vault 1.12 is used to create the storage entry. If Vault say, 1.9 was used and later upgraded to 1.12, this test would fail, because the 1.9 entry wouldn't have default values added where empty strings were. 


I'd prefer to back this particular change out, but leave the `storage.go` change you have.

Reply:
I'll go ahead and back this part of the change out, but I do believe it would have resolved the issue (had we not also returned the default when expiry was `""`). My reasoning is that the `ParseDuration` call happens after the check for the CRL being enabled. We have the CRL disabled, and therefore would have returned out of this method before the duration was parsed.

Reply:
Setting the default is definitely simpler and has way less implications on the remaining code. I definitely have less experience (a.k.a. no experience) on the codebase. I really appreciate the time you've spent on this (and your thorough explanations), thank you!